File: C/beginner/1985.c
// https://judge.beecrowd.com/pt/problems/view/1985
// MacPRONALTS


// Libraries
#include <stdio.h>

// Main
int main(){
    int productId, productsQtOrder, productsTotalQt;
    float productsTotalPrice = 0;

    scanf("%d", &productsTotalQt);

    for (int i = 0; i < productsTotalQt; i++){
        scanf("%d %d", &productId, &productsQtOrder);

        switch (productId){
            case 1001:
                productsTotalPrice += productsQtOrder * 1.50;
                break;
            case 1002:
                productsTotalPrice += productsQtOrder * 2.50;
                break;
            case 1003:
                productsTotalPrice += productsQtOrder * 3.50;
                break;
            case 1004:
                productsTotalPrice += productsQtOrder * 4.50;
                break;
            case 1005:
                productsTotalPrice += productsQtOrder * 5.50;
                break;
            default:
                productsTotalPrice += 0;
                break;
        }
    }

    printf("%.2f\n", productsTotalPrice);

    return 0;
}

// By: Fernando Serra
// https://github.com/fernandosserra

File: Haskell/beginner/1985.hs
-- https://judge.beecrowd.com/pt/problems/view/1985
-- MacPRONALTS

import Text.Printf (printf)
import Control.Monad (replicateM)

productPrices :: [(Int, Float)]
productPrices =
    [ (1001, 1.50)
    , (1002, 2.50)
    , (1003, 3.50)
    , (1004, 4.50)
    , (1005, 5.50)
    ]


calculateItemPrice :: Int -> Int -> Float
calculateItemPrice productId productsQtOrder =
    case lookup productId productPrices of
        Just price -> fromIntegral productsQtOrder * price
        Nothing    -> 0.0


main :: IO ()
main = do    
    productsTotalQtStr <- getLine
    let productsTotalQt = read productsTotalQtStr :: Int
    
    itemPrices <- replicateM productsTotalQt $ do
        line <- getLine
        let [productId, productsQtOrder] = map read (words line) :: [Int]
        return $ calculateItemPrice productId productsQtOrder

    let productsTotalPrice = sum itemPrices
    printf "%.2f\n" productsTotalPrice

-- By: Fernando Serra & Washu Sama
-- https://github.com/fernandosserra